### PR TITLE
Fixes #14929: Add support of semicolon-less JS to asset/compress comm…

### LINKED ISSUE #14929: asset/compress needs semicolon-less JavaScripts support
New released yii-pjax 2.0.7 breaks my compressed JavaScript which generated by `asset/compress` command. JS files are joined with no separators after `jquery.pjax.js`.

### What steps will reproduce the problem?
Create an Yii project, then simply use `yii asset/compress`.

### What is the expected result?
At least a semicolon should be inserted. I prefer wrapping each scripts by `function(){ ... }();`.

### What do you get instead?
no-semicolon

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.12
| PHP version      | 7.0.15, 7.1.8
| Operating system |Ubuntu, MacOS

### PR DESCRIPTION
**Pull request by tanakahisateru**
My idea to fix #14929 by wrapping with function scope block.

…and.

| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #14929

### CODE REVIEW

**Comment by shirase** [comment]
Global variable will be undefined beyond wrapper function context.
```
var test =""; // is global, but
(function() {
var test =""; // is not global
})();
```

**Comment by tanakahisateru** [comment]
To insert a ";" doesn't break BC. Should I change?

**Comment by tanakahisateru** [comment]
I'm sure that wrapping was too aggressive. I will change this PR to avoid function.

**Comment by samdark** [comment]
Isn't https://github.com/yiisoft/jquery-pjax/commit/0d4cc3f479d4b7b800aefca1275a852c8d3dd3c6 fixing the problem?

File: framework/console/controllers/AssetController.php [hunk 1]
```diff
@@ -569,7 +569,9 @@ public function combineJsFiles($inputFiles, $outputFile)
         $content = '';
         foreach ($inputFiles as $file) {
             $content .= "/*** BEGIN FILE: $file ***/\n"
+                . "(function(){\n"
                 . file_get_contents($file)
```
Comment (samdark) [inline framework/console/controllers/AssetController.php]: It's better to check if trimmed contents end with `;` and add it if it's not there.

**Review by samdark — CHANGES_REQUESTED** [review]

**Comment by tanakahisateru** [comment]
@samdark I've modified the commit.

**Review by rob006 — APPROVED** [review]
Changelog entry is missing.

**Review by samdark — APPROVED** [review]

**Pull request merged**